website/_kb/user-guide/querying/query-results-data-source.md

```diff
@@ -5,34 +5,30 @@ helpscout_url: https://help.redash.io/article/152-query-results-data-source
     FROM query_123 a 
     JOIN query_456 b ON a.id = b.id
-    
 
-When referencing a query as a table in the Query Results query, the table name
-is the query ID (the number in the URL) prefixed with _query_. So for
-example, a query with the URL
-`https://app.redash.io/acme/queries/49588/source` will become `query_49588`
-when referenced as a table.
+Each of your existing queries constitutes its own "table" to SQLite. The table name is the string `query_` concatenated with your desired Query ID, which can be found in that query's URL. So for example, a query with the URL `https://app.redash.io/acme/queries/49588/source` will have the table name `query_49588` in SQLite.
 
-{% callout %}
-You need to make sure the table name (`query_...`) is on the same line as the
-`FROM`/`JOIN` keywords.
+{% callout warning %}
+Be careful that your table name (`query_49588` above) appears on the same line as the associated `FROM` and `JOIN` keywords. Redash can't parse it correctly otherwise.
 {% endcallout %}
 
-Few notes:
+### Cached Query Results
+When you query the **Query Results** Data Source, Redash executes the underlying queries first. This ensures you have recent results in the event that you [schedule this query](/help/user-guide/querying/scheduling-a-query). You can reduce the running time of **Query Results** queries by using `cached_query_` for your table names instead of `query_`. This tells Redash to use the cached results from the most recent execution of a given query. This reduces the number of calls to your underlying Data Sources, improving performance by using older data. Here's an example query:
 
-  1. When you run a query, we execute the underlying queries as well to make sure you have recent results in case you schedule this query. If you want to reduce the number of times you run the same query you can use `cached_query_...` instead `query_...`
-  2. The processing of the data is being done by SQLite in memory - in case of large result sets it might fail due to memory running out.
-  3. Access to the data source is governed by the groups it's associated with, like any other data source. When a user runs a query we also check if he has permission to execute queries on the data sources the original queries use. So while a user who has access to this data source will be able to see any query that uses it, he won't be able to execute queries on data sources he doesn't have access to.
+    SELECT a.name, b.count 
+    FROM cached_query_123 a 
+    JOIN cached_query_456 b ON a.id = b.id
 
+### Query Results Permissions
+Access to the **Query Results** Data Source is governed by the groups it's associated with, [like any other Data Source](/help/user-guide/users/permissions-groups). But Redash will also check if that user has permission to execute queries on the Data Sources the original queries use. In this way the **Query Results** Data Source will not allow a user to circumvent restrictions established elsewhere.
```

Comment:
Worth mentioning that the user will still be able to see a cached result of this Query Results query, but won't be able to refresh it.